src/textual/dom.py

```diff
@@ -259,7 +259,7 @@ def screen(self) -> "Screen":
         from .screen import Screen
 
         node = self
-        while node and not isinstance(node, Screen):
+        while node is not None and not isinstance(node, Screen):
```

Comment:
This is the line that ultimately resulted in those weird `NoScreen` errors. Because ListView has a __len__, the condition was evaluating and terminating the search for the Screen.

Reply:
Ah. Good catch!